File: C/20_Divide&Conquer/BOJ_11728_배열합치기.c
#include <stdio.h>
main() {
	int n, m;
	scanf("%d %d", &n, &m);
	int* a = (int*)malloc(sizeof(int) * n);
	int* b = (int*)malloc(sizeof(int) * m);
	int* arr = (int*)malloc(sizeof(int) * (n + m));
	for (int i = 0; i < n; i++) {
		scanf("%d", &a[i]);
	}
	for (int i = 0; i < m; i++) {
		scanf("%d", &b[i]);
	}
	int i = 0, j = 0, cnt = 0;
	while (i < n && j < m) {
		if (a[i] < b[j]) {
			arr[cnt++] = a[i++];
		}
		else {
			arr[cnt++] = b[j++];
		}
	}
	while (i < n) arr[cnt++] = a[i++];
	while (j < m) arr[cnt++] = b[j++];
	for (int i = 0; i < cnt; i++) {
		printf("%d ", arr[i]);
	}
}

File: JAVA/20_Divide&Conquer/BOJ_11728_배열합치기.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        int[] a = new int[n];
        int[] b = new int[m];
        int[] arr = new int[n + m];
        st = new StringTokenizer(br.readLine(), " ");
        for (int i = 0; i < n; i++) {
            a[i] = Integer.parseInt(st.nextToken());
        }
        st = new StringTokenizer(br.readLine(), " ");
        for (int i = 0; i < m; i++) {
            b[i] = Integer.parseInt(st.nextToken());
        }
        int i = 0, j = 0, cnt = 0;
        while (i < n && j < m) {
            if (a[i] < b[j]) {
                arr[cnt++] = a[i++];
            } else {
                arr[cnt++] = b[j++];
            }
        }
        while (i < n) arr[cnt++] = a[i++];
        while (j < m) arr[cnt++] = b[j++];
        StringBuilder sb = new StringBuilder();
        for (int e : arr) {
            sb.append(e).append(" ");
        }
        System.out.println(sb);
    }
}